### DISCUSSION TITLE
请问下，如何理解dumi为组件研发而生`？

### DISCUSSION THREAD

**Discussion by liaodalin19903** [umijs/dumi#1474] [Q&A] [answered]
请问下，如何理解dumi为组件研发而生`？

是指的，dumi主要功能中有：可以直接依据已经做好的代码现场展示出UI效果是吗？而不是指的对新的组件研发有辅助功能。
![image](https://user-images.githubusercontent.com/12727582/218376968-12617460-d851-4772-a414-ca25bf8319e9.png)

**Comment by PeachScript** [comment, marked as answer]
是指为 DDD（Demo-Driven Development or Documentation-Driven Development）的研发模式提供相关能力

如果没有 dumi 或类似工具，开发者需要创造额外的运行环境去查看组件是否如预期工作，也需要创造额外的应用去给用户提供使用文档
